src/main/java/htsjdk/samtools/cram/structure/BlockCompressionMethod.java

```diff
@@ -18,5 +18,23 @@
 package htsjdk.samtools.cram.structure;
 
 public enum BlockCompressionMethod {
-    RAW, GZIP, BZIP2, LZMA, RANS
+    RAW(0),
+    GZIP(1),
+    BZIP2(2),
+    LZMA(3),
+    RANS(4);
+
+    private final int specValue;
+
+    /**
+     * The block compression methods specified by Section 8 of the CRAM spec
+     * @param value the number assigned to each block compression method in the CRAM spec
+     */
+    BlockCompressionMethod(final int value) {
```

Comment:
The naming of these is inconsistent and "spec" seems a bit strange in the name.  I would drop spec and just go with `getId` or `getTypeId` as appropriate.  The javadoc is a good addition, it would make sense to put it as class level javadoc  and on the getter, as well since users will never actually see the enum constructor.